components/metrics/metrics_slots.go

```diff
@@ -0,0 +1,137 @@
+var SlotMetrics = collector.NewCollection(slotNamespace,
+	collector.WithMetric(collector.NewMetric(totalBlocks,
+		collector.WithType(collector.Counter),
+		collector.WithLabels(slotLabelName),
+		collector.WithPruningDelay(10*time.Minute),
+		collector.WithHelp("Number of blocks seen by the node in a slot."),
+		collector.WithInitFunc(func() {
+			deps.Protocol.Events.Engine.BlockDAG.BlockAttached.Hook(func(block *blocks.Block) {
+				eventSlot := int(block.ID().Index())
+				deps.Collector.Increment(slotNamespace, totalBlocks, strconv.Itoa(eventSlot))
+			}, event.WithWorkerPool(Component.WorkerPool))
+		}),
+	)),
+
+	collector.WithMetric(collector.NewMetric(acceptedBlocksInSlot,
+		collector.WithType(collector.Counter),
+		collector.WithHelp("Number of accepted blocks in a slot."),
+		collector.WithLabels(slotLabelName),
+		collector.WithPruningDelay(10*time.Minute),
+		collector.WithInitFunc(func() {
+			deps.Protocol.Events.Engine.BlockGadget.BlockAccepted.Hook(func(block *blocks.Block) {
+				eventSlot := int(block.ID().Index())
+				deps.Collector.Increment(slotNamespace, acceptedBlocksInSlot, strconv.Itoa(eventSlot))
+			}, event.WithWorkerPool(Component.WorkerPool))
+		}),
+	)),
+	collector.WithMetric(collector.NewMetric(invalidBlocks,
+		collector.WithType(collector.Counter),
+		collector.WithLabels(slotLabelName),
+		collector.WithPruningDelay(10*time.Minute),
+		collector.WithHelp("Number of invalid blocks in a slot."),
+		collector.WithInitFunc(func() {
+			deps.Protocol.Events.Engine.BlockDAG.BlockInvalid.Hook(func(block *blocks.Block, err error) {
+				eventSlot := int(block.ID().Index())
+				deps.Collector.Increment(slotNamespace, invalidBlocks, strconv.Itoa(eventSlot))
+			}, event.WithWorkerPool(Component.WorkerPool))
+		}),
+	)),
+	collector.WithMetric(collector.NewMetric(acceptedAttachments,
+		collector.WithType(collector.Counter),
+		collector.WithLabels(slotLabelName),
+		collector.WithPruningDelay(10*time.Minute),
+		collector.WithHelp("Number of accepted attachments by the node per slot."),
+		collector.WithInitFunc(func() {
+			deps.Protocol.MainEngineInstance().Ledger.OnTransactionAttached(func(transactionMetadata mempool.TransactionMetadata) {
+				transactionMetadata.OnAccepted(func() {
+					for _, attachmentBlockID := range transactionMetadata.Attachments() {
+						if block, exists := deps.Protocol.MainEngineInstance().BlockCache.Block(attachmentBlockID); exists && block.IsAccepted() {
+							deps.Collector.Increment(slotNamespace, acceptedAttachments, strconv.Itoa(int(attachmentBlockID.Index())))
+						}
+					}
+				})
+			})
+		}),
+	)),
+	collector.WithMetric(collector.NewMetric(createdConflicts,
+		collector.WithType(collector.Counter),
+		collector.WithLabels(slotLabelName),
+		collector.WithPruningDelay(10*time.Minute),
+		collector.WithHelp("Number of conflicts created per slot."),
+		collector.WithInitFunc(func() {
+			deps.Protocol.Events.Engine.BlockDAG.BlockAttached.Hook(func(block *blocks.Block) {
```

Comment:
Sorry, I don't get why we attach to the blockAttached here?
Is that to initialize the label for that slot with 0?
Wouldn't that overwrite the conflicts value with 0 with every new attached block in that slot?
Can we add a comment here in the code that explains what this is supposed to do?